src/libAtomVM/context.c

```diff
@@ -774,15 +775,15 @@ static struct Monitor *context_monitors_handle_terminate(Context *ctx)
                 Context *target = globalcontext_get_process_nolock(glb, local_process_id);
                 // Target cannot be NULL as we processed Demonitor signals
                 assert(target != NULL);
-                int required_terms = REF_SIZE + TUPLE_SIZE(5);
+                int required_terms = TERM_BOXED_PROCESS_REF_SIZE + TUPLE_SIZE(5);
                 if (UNLIKELY(memory_ensure_free(ctx, required_terms) != MEMORY_GC_OK)) {
                     // TODO: handle out of memory here
                     fprintf(stderr, "Cannot handle out of memory.\n");
                     globalcontext_get_process_unlock(glb, target);
                     AVM_ABORT();
                 }
                 // Prepare the message on ctx's heap which will be freed afterwards.
-                term ref = term_from_ref_ticks(monitored_monitor->ref_ticks, &ctx->heap);
+                term ref = term_make_process_reference(target->process_id, monitored_monitor->ref_ticks, &ctx->heap);
```

Comment:
This is not what OTP does and there probably is a good reason.
Regular monitors are smaller than aliases.
```
7> Pid = spawn(fun() -> receive ok -> ok end end).
<0.94.0>
8> monitor(process, Pid).
#Ref<0.3008598808.1200095248.57512>
9> monitor(process, Pid, [{alias, explicit_unalias}]).
#Ref<0.0.11651.3008598808.1200160784.57533>
10> make_ref().
#Ref<0.3008598808.1200095248.57553>
```